File: medium/442-Find_All_Duplicates_in_an_array.cpp
vector<int> findDuplicates(vector<int>& nums) {
	vector<int> ans;
	
	for(int a = 0; a < nums.size(); a++) {
		int value = abs(nums[a]);
		if(nums[value-1] < 0) {
			ans.push_back(value);
		}
		else {
			nums[value-1] = -nums[value-1];
		}
	}
	return ans;
}

File: medium/442-Find_All_Duplicates_in_an_array.java
public List<Integer> findDuplicates(int[] nums) {
	List<Integer> ans = new ArrayList<>();
	for(int a = 0; a < nums.length; a++) {
		int value = Math.abs(nums[a]);
		if(nums[value-1] < 0) {
			ans.add(value);
		}
		else {
			nums[value-1] = -nums[value-1];
		}
	}
	return ans;
}